### PR TITLE
Remove --pathfinding-eth-address option

### LINKED ISSUE #4054: Wrong type for pathfinding_eth_address in run_app
During #4050 I found that the parameter `pathfinding_eth_address` is currently a str, whereas it should be `Address` to match the convention.

https://github.com/raiden-network/raiden/pull/4050/files#diff-4969908fb9333b5d556f203dcda92e1dR140

### PR DESCRIPTION
**Pull request by jomuel**
Now that the PFS includes its eth address in the `/info` endpoint response, we don't need a cli and config option to configure it anymore.

Closes #4054

### CODE REVIEW

**Comment by palango** [comment]
See https://github.com/raiden-network/raiden/issues/4054

**Comment by Dominik1999** [comment]
is this now automatically a checksum address?

**Comment by palango** [comment]
> is this now automatically a checksum address?

I don't think so.

**Comment by jomuel** [comment]
> is this now automatically a checksum address?

Yes, it will accept only EIP55 addresses for this parameter. Same as for all the contract addresses.

**Comment by palango** [comment]
The CLI parameter yes, but not the stuff send in the IOU, right?

**Comment by jomuel** [comment]
> The CLI parameter yes, but not the stuff send in the IOU, right?

Isn't the stuff send in the IOU the config entry, aka cli parameter?

File: raiden/ui/cli.py [hunk 1]
```diff
@@ -259,7 +259,7 @@ def options(func):
                     'manually, i. e. "--pathfinding-service-address" set to a value other than '
                     '"auto", this argument must be set to a valid EIP55 address.'
                 ),
-                type=str,
+                type=ADDRESS_TYPE,
```
Comment (palango) [inline raiden/ui/cli.py]: This converts the argument to `Address`, see `AddressType`, so the argument in `run_app` needs change.

**Comment by ulope** [comment]
AFAIK since raiden-network/raiden-services#313 we don't really need this parameter anymore. Someone would just need to implement using the address from the info response.

**Comment by palango** [comment]
@ulope Good point! @jomuel , do you have time to fix that as well?

**Comment by jomuel** [comment]
> @ulope Good point! @jomuel , do you have time to fix that as well?

Sure.

File: raiden/network/pathfinding.py [hunk 2]
```diff
@@ -162,6 +157,19 @@ def configure_pfs_or_exit(
         )
         sys.exit(1)
     else:
+        if "payment_address" not in pathfinding_service_info:
```
Comment (palango) [inline raiden/network/pathfinding.py]: We could only check that when the PFS' fee is higher that 0, but as we always send the address this is not important.

File: raiden/tests/scenarios/pfs1_get_a_simple_path.yaml [hunk 3]
```diff
@@ -24,7 +24,6 @@ nodes:
     environment-type: development
     routing-mode: pfs
     pathfinding-max-paths: 5
-    pathfinding-eth-address: "0x0bae0289aaa26845224f528f9b9defe69e01606e"
```
Comment (palango) [inline raiden/tests/scenarios/pfs1_get_a_simple_path.yaml]: Nice, I didn't even remember those were here.

File: raiden/network/pathfinding.py [hunk 4]
```diff
@@ -176,11 +180,7 @@ def configure_pfs_or_exit(
         click.secho(msg)
         log.info("Using PFS", pfs_info=pathfinding_service_info)
 
-    return PFSConfiguration(
-        url=pfs_address,
-        eth_address=pfs_eth_address,
-        fee=pathfinding_service_info.get("price_info", 0),
-    )
+    return PFSConfiguration(url=pfs_address, eth_address=pfs_eth_address, fee=fee)
```
Comment (palango) [inline raiden/network/pathfinding.py]: With the fee > 0 changes, you should make `PFSConfiguration.eth_address` `Optional`.

**Pull request merged**